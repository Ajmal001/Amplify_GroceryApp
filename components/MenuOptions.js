import * as React from "react";
import { View } from "react-native";
import { Button, Menu, Divider, Provider } from "react-native-paper";
import RoundButton from "./RoundButton";

const MenuOptions = ({ actionsMenu, groceryList }) => {
  const [visible, setVisible] = React.useState(false);

  const openMenu = () => setVisible(true);

  const closeMenu = () => setVisible(false);

  return (
    <View
      style={{
        flexDirection: "row",
        justifyContent: "center",
      }}
    >
      <Menu
        visible={visible}
        onDismiss={closeMenu}
        anchor={
          <RoundButton
            onPress={openMenu}
            name="dots-vertical"
            style={{ marginRight: 20 }}
          />
        }
      >
        {actionsMenu.map((action, index) => (
          <Menu.Item
            key={index}
            icon={action.icon}
            onPress={() => {
              action.onPress(groceryList);
              closeMenu();
            }}
            title={action.title}
          />
        ))}
      </Menu>
    </View>
  );
};

export default MenuOptions;
