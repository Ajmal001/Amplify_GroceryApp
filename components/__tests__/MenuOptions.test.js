import React from 'react';
import renderer from 'react-test-renderer';
import MenuOptions from '../MenuOptions';

jest.mock("@expo/vector-icons", () => {
    return {
      MaterialCommunityIcons: () => {
        return <></>;
      },
    };
  });
  
test('renders correctly', () => {
  const tree = renderer.create(<MenuOptions
    actionsMenu={[]}
    groceryList={''}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});