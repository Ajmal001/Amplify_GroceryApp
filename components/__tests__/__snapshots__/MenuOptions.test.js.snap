// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    collapsable={false}
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 20,
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
